File: Modules/Common/Techs/SIE.Web.Tech/Processs/Commands/ProcessSkillCommand.js
SIE.defineCommand('SIE.Web.Tech.Processs.Commands.ProcessSkillCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'SkillId', targetClassName: 'SIE.Resources.Skills.Skill' },
    },
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    canExecute: function (view) {     
        var p = view.getParent();
        if (!p) { return true; }
        var entity = p.getCurrent();
        if (entity === null) { return false };
        return entity.dirty == false;//entity.data.CreateBy > 0;
    },
    save: function (win) {
        var me = this;
        var indata = {};
        var selections = this._targetView.getSelection();
        if (selections && selections.length > 0) {
            var operationDatas = [];
            SIE.each(selections, function (item) {
                var skillId = item.getId();
                if (me._sourceViewSelectItems.indexOf(skillId) === -1) {
                    var skill = { ProcessId: me._sourceId, SkillId: skillId };
                    operationDatas.push(skill);
                }
            });
            indata = operationDatas;
            me._targetView.execute({
                data: indata,
                success: function (res) {
                    win.close();
                    me._ownerView.loadChildData(true);
                }
            }, me._ownerView);
        }
        else {
            SIE.Msg.showWarning('没有可提交的数据'.t());
        }
    },
});

File: Modules/Common/Techs/SIE.Web.Tech/Processs/Commands/ProcessSkillCommand.cs
using SIE.Domain;
using SIE.Tech.Processs;
using SIE.Web.Command;
using System;
using System.Collections.Generic;

namespace SIE.Web.Tech.Processs.Commands
{
    /// <summary>
    /// 工序技能命令
    /// </summary>
    [JsCommand("SIE.Web.Tech.Processs.Commands.ProcessSkillCommand")]
    public class ProcessSkillCommand : ViewCommand
    {
        /// <summary>
        /// 工序技能命令执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var meta = ClientEntities.Find(args.Type);
            var savedData = RF.Find(meta.EntityType).NewList();
            var skillList = args.Data.ToJsonObject<List<ProcessSkill>>();
            Check.NotNullOrEmpty(skillList, nameof(skillList));
            if (skillList == null || skillList.Count == 0)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(skillList)));
            }

            foreach (var item in skillList)
            {
                var skill = new ProcessSkill();
                skill.SkillId = item.SkillId;
                skill.ProcessId = item.ProcessId;
                savedData.Add(skill);
            }

            RF.Save(savedData);
            return true;
        }
    }
}